fortex/health/processors/scispacy_processor.py

```diff
@@ -0,0 +1,168 @@
+# Copyright 2022 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+SciSpacy Processor
+"""
+from typing import Dict, Set
+import importlib
+
+import spacy
+from forte.common import Resources
+from forte.common.configuration import Config
+from forte.data.data_pack import DataPack
+from forte.processors.base import PackProcessor
+
+# pylint: disable=unused-import
+from scispacy.abbreviation import AbbreviationDetector
+from scispacy.hyponym_detector import HyponymDetector
+
+from ftx.medical.clinical_ontology import Hyponym, Abbreviation, Phrase
+
+__all__ = [
+    "ScispaCyProcessor",
+]
+
+
+class ScispaCyProcessor(PackProcessor):
+    r"""
+    Implementation of this ScispaCyProcessor has been based on ScispaCy
+    pretained models (based on spacy),  A rendition of it that
+    exists on github has been referred to as well.
+
+    Referred repository link:
+    https://pythonlang.dev/repo/allenai-scispacy/
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.extractor = None
+
+    def set_up(self):
+        if self.configs.require_gpu:
+            spacy.require_gpu(self.configs.gpu_id)
+        if self.configs.prefer_gpu:
+            spacy.prefer_gpu(self.configs.gpu_id)
+        self.extractor = spacy.load(self.configs.model_name)
+        if self.configs.pipe_name == "abbreviation_detector":
+            self.extractor.add_pipe(self.configs.pipe_name)
+        else:  # hyponym
```

Comment:
We can add an elif instead of else, so we don't enter this condition due to typos and for future addition of pipe_names. Also, lets add comments on new lines instead of just appending to code on the same line, just for better readability.